src/config/middleware/logging-middleware.ts

```diff
@@ -0,0 +1,67 @@
+import {MomentoLogger, MomentoLoggerFactory} from '../logging/momento-logger';
+import {Middleware, MiddlewareRequestHandler} from './middleware';
+import {Metadata, StatusObject} from '@grpc/grpc-js';
+import {Message} from 'google-protobuf';
+
+export class LoggingMiddlewareRequestHandler
+  implements MiddlewareRequestHandler
+{
+  private readonly logger: MomentoLogger;
+  constructor(logger: MomentoLogger) {
+    this.logger = logger;
+  }
+
+  onRequestMetadata(metadata: Metadata): Promise<Metadata> {
+    this.logger.info(
```

Comment:
Is this how we will implement trace level logging?

Reply:
yeah that's what i'm thinking, and i may do it in this pr